### ISSUE TITLE
 Can not disable a channel from an item with custom name

### ISSUE THREAD

**Issue by emiovska** [eclipse/smarthome#2234]
In the demo.items there is an item with a custom name:
`Number Weather_Temperature 		"Outside Temperature [%.1f °C]"	<temperature> (Weather_Chart) { channel="yahooweather:weather:berlin:temperature" }`

When I try to disable the temperature channel from this item I receive a toastr with the following message: _ERROR: 404 - Not Found._ The response contains the following error: 
`"error": {
    "message": "Link yahooweather_weather_berlin_temperature -\u003e yahooweather:weather:berlin:temperature does not exist!",
    "http-code": 404
  }`

According to the message of the error, it seems that the name of the item is wrong. The name of the item is **Weather_Temperature** but the API received **yahooweather_weather_berlin_temperature**.

I started to investigate the problem and I found that in the _$scope.disableChannel_ function from the _controllers.configuration.js_ file, the name of the item is sending like: 
**thing.UID.replace(/[^a-zA-Z0-9_]/g, "_") + '_' + channelID.replace(/[^a-zA-Z0-9_]/g, "_")**, 
which is not the case when we have an item with custom name. Also, in the same function there is one unused variable **linkedItem** which I think is the key for solving the issue.

**Comment by kaikreuzer** [comment]
@aounhaider1 Could you have a look?

**Comment by kaikreuzer** [comment]
@emiovska Please note that even if the bug is fixed, you won't be able to delete the link, since it is provided through the items file (and thus read-only). But the response should indeed be a 405, not a 404.

**Comment by emiovska** [comment]
Yes @kaikreuzer , you are totally right. I think that I should change the name of the issue to be "Wrong error message when I try to disable a channel from a read-only item with custom name". What do you think?

**Comment by kaikreuzer** [comment]
sure, that would fit better.

**Comment by aounhaider1** [comment]
The server currently responds with a 404 in the above scenario. It needs to be changed on the server side.

**Comment by emiovska** [comment]
> The response contains the following error: 
"error": { "message": "Link yahooweather_weather_berlin_temperature -\u003e yahooweather:weather:berlin:temperature does not exist!", "http-code": 404 }

According to the response error I think that the server response with status 404, because receives wrong item name.  The name of the item is Weather_Temperature but the server received yahooweather_weather_berlin_temperature.

>  in the $scope.disableChannel function from the controllers.configuration.js file, the name of the item is sending like: 
thing.UID.replace(/[^a-zA-Z0-9_]/g, "") + '' + channelID.replace(/[^a-zA-Z0-9_]/g, "_"), 
which is not the case when we have an item with custom name. Also, in the same function there is one unused variable linkedItem which I think is the key for solving the issue.

Because the advanceMode is false,  I think that the name of the item maybe should be sent like _channel.linkedItems[0]_,

**Comment by aounhaider1** [comment]
````
The name of the item is Weather_Temperature but the server received 
````
Which means that item was linked in advanced mode, as simple mode always generates itemName by replacing semicolons with underscore.
Moreover, with https://github.com/eclipse/smarthome/pull/2195 we can not just delete the first linked item as there can be multiple linked items now.
Are you sure you are on latest master? There was recently a bug (https://github.com/eclipse/smarthome/pull/2173) that caused the linking mode to not change as expected on Save.

**Comment by emiovska** [comment]
I have the latest version of the master and the item name is manually set in the demo.items file
